pkg/storer/internal/reserve/reserve.go

```diff
@@ -40,6 +41,8 @@ type Reserve struct {
 	capacity int
 	size     atomic.Int64
 	radius   atomic.Uint32
+
+	binIDMtx sync.Mutex
```

Comment:
Not sure what fields this mutex is guarding?

Reply:
https://github.com/ethersphere/bee/pull/4199/files#diff-e37eed2ab5dfe1533be0257473f5b76738361f02d2be71da1b7bf4d5216a00c7R395